IPIP/0000-gateway-json-cbor-response-format.md

```diff
@@ -0,0 +1,79 @@
+# IPIP 0000: Gateway JSON and TAR Response Formats
+
+- Start Date: 2022-10-07
+- Related Issues:
+  - [ipfs/in-web-browsers/issues/182]
+  - [ipfs/specs/pull/328]
+  - [ipfs/kubo/issues/8823]
+  - [ipfs/kubo/pull/9335]
+  - [ipfs/go-ipfs/issues/7552]
+
+## Summary
+
+Add [DAG-JSON], [DAG-CBOR], JSON and CBOR response formats to the [HTTP Gateway](../http-gateways/).
+
+## Motivation
+
+Currently, the gateway supports requesting data in the [DAG-PB], RAW, and [CAR]
+formats. However, it should be possible to download everything over the gateway
+in order to unlock the potentials of the [IPLD Data Model][ipld-data-model].
+
+For example, DAG-PB has a [logical format][dag-pb-format] which makes it possible
+to represent a DAG-PB directory as a [DAG-JSON] document. This means that, if we
+support DAG-JSON in the gateway, then we would support
+[JSON responses for directory listings][ipfs/go-ipfs/issues/7552], which has been
+requested by our users in the past.
+
+In addition, this functionality is already present on the current Kubo CLI. By
+bringing it to the gateways, we provide users with more power when it comes
+to traversing IPLD data.
+
+## Detailed design
+
+The solution is to allow the Gateway to support serializing an IPLD model
+representation as [DAG-JSON], [DAG-CBOR], JSON and CBOR by requesting them
+using either the `Accept` HTTP header or the `format` URL query.
+
+## Test fixtures
+
+TODO
+
+## Design rationale
+
+The current gateway already supports different response formats via the
+`Accept` HTTP header and the `format` URL query. This IPIP proposes adding
+one more supported format to that list.
+
+### User benefit
+
+The user benefits from this change as they will now be able to retrieve
+content encoded in the traversable DAG-JSON and DAG-CBOR formats. This is
+something that has been [requested before][ipfs/go-ipfs/issues/7552].
+
+### Compatibility
+
+This IPIP is backwards compatible.
+
+### Security
+
+TODO
```

Comment:
Mention security considerations around using serializer/deserializer that follows RFC/specs:
- JSON – link to  RFC 8259, Section 12.
- CBOR – link to  Section 10 of RFC 8949
- DAG-JSON/CBOR –  same as non-DAG version + note these are more strict variants that need to follow IPLD specs and error when payload is not strict enough  (linking to specs should be enough)